src/engraving/infrastructure/rtti.h

```diff
@@ -37,13 +39,32 @@ bool is_classof(From* p) noexcept
     return p ? ResultType::classof(p) : false;
 }
 
+enum class CastMode {
+    CHECK = 0,  // should be correct; assert if still not
+    BAD,        // known to be bad; don’t assert, we’ll fix it later
+    MAYBE_BAD   // in this case it’s okay if type mismatch
+};
+
 template<typename To, typename From>
-auto item_cast(From* p) noexcept
+To item_cast(From from, CastMode mode = CastMode::CHECK) noexcept
 {
-    using ResultType = std::remove_cv_t<std::remove_pointer_t<To> >;
-    return is_classof<ResultType>(p) ? static_cast<To>(p) : nullptr;
+    if constexpr (mu::runtime::isDebug()) {
+        if constexpr (std::is_same_v<From, void>) {
+            return static_cast<To>(from);
+        } else {
+            To casted = dynamic_cast<To>(from);
+            if (mode == CastMode::CHECK) {
+                assert(casted || !from);
+            }
+            return casted;
+        }
+    } else {
+        return static_cast<To>(from);
```

Comment:
We can: 
* Add static check (based on classof)  
* Always use dynamic_cast 

But I'm not sure if this is necessary. I think we can keep it this way, detect all errors in the debug, and have maximum performance in the release

Reply:
There is a bad cast there, so we can't just keep static_cast
I made it so that the dynamic cast was used.